include/xrpl/protocol/TxFlags.h

```diff
@@ -130,6 +132,23 @@ constexpr std::uint32_t const tfOnlyXRP                    = 0x00000002;
 constexpr std::uint32_t const tfTrustLine                  = 0x00000004;
 constexpr std::uint32_t const tfTransferable               = 0x00000008;
 
+// MPTokenIssuanceCreate flags:
+// NOTE - there is intentionally no flag here for 0x01 because that
+// corresponds to lsfMPTLocked, which this transaction cannot mutate. 
+constexpr std::uint32_t const tfMPTCanLock                 = lsfMPTCanLock;
+constexpr std::uint32_t const tfMPTRequireAuth             = lsfMPTRequireAuth;
+constexpr std::uint32_t const tfMPTCanEscrow               = lsfMPTCanEscrow;
+constexpr std::uint32_t const tfMPTCanTrade                = lsfMPTCanTrade;
+constexpr std::uint32_t const tfMPTCanTransfer             = lsfMPTCanTransfer;
+constexpr std::uint32_t const tfMPTCanClawback             = lsfMPTCanClawback;
+
+// MPTokenAuthorize flags:
+constexpr std::uint32_t const tfMPTUnauthorize             = 0x00000001;
```

Comment:
@shawnxie999 I'm not going to die on this hill, but I see `MPTokenAuthorize` as more analogous to [DepositPreauth](https://xrpl.org/docs/references/protocol/transactions/types/depositpreauth) combined with [TrustSet](https://xrpl.org/docs/references/protocol/transactions/types/trustset).

It's not an exact comparison because `DepositPreauth` uses fields (`Authorize` and `Unauthorize`) instead of flags, but that controls the ability to send funds, not hold them. Meanwhile `TrustSet` uses a `tfSetfAuth` flag _that cannot be cleared_, which does control the ability to hold tokens.

OTOH, `NFTokenSellOffer` is more comparable to `OfferCreate`, which has the `tfSell` flag. Since `OfferCreate` came well before `NFTokenSellOffer`, that's probably the reason for that design decision. Also, the consequences of getting that flag wrong are much less significant that unintentionally authorizing, say, a sanctioned entity to hold your MPToken.

So where does that leave us?

I don't think there's a clear best answer. I don't have any empirical data, but my gut / experience is telling me that forgetting to set the flag is more likely than using the wrong flag value for authorize vs. unauthorize. I think making the change is a good idea, but I'm not going to require it.

Reply:
Thank you for the feedback. I think I'd prefer leaving the current design as it is.